test-suite-data.json

```diff
@@ -466,5 +466,77 @@
     "qualifiers": null,
     "subpath": null,
     "is_invalid": true
+  },
+  {
+    "description": "minimal Hugging Face model",
+    "purl": "pkg:huggingface/distilbert-base-uncased@043235d6088ecd3dd5fb5ca3592b6913fd516027",
+    "canonical_purl": "pkg:huggingface/distilbert-base-uncased@043235d6088ecd3dd5fb5ca3592b6913fd516027",
+    "type": "huggingface",
+    "namespace": null,
+    "name": "distilbert-base-uncased",
+    "version": "043235d6088ecd3dd5fb5ca3592b6913fd516027",
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "Hugging Face model with staging endpoint",
+    "purl": "pkg:huggingface/microsoft/deberta-v3-base@559062ad13d311b87b2c455e67dcd5f1c8f65111?repository_url=https://hub-ci.huggingface.co",
+    "canonical_purl": "pkg:huggingface/microsoft/deberta-v3-base@559062ad13d311b87b2c455e67dcd5f1c8f65111?repository_url=https://hub-ci.huggingface.co",
+    "type": "huggingface",
+    "namespace": "microsoft",
+    "name": "deberta-v3-base",
+    "version": "559062ad13d311b87b2c455e67dcd5f1c8f65111",
+    "qualifiers": {"repository_url": "https://hub-ci.huggingface.co"},
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "Hugging Face model with various cases",
+    "purl": "pkg:huggingface/EleutherAI/gpt-neo-1.3B@797174552AE47F449AB70B684CABCB6603E5E85E",
```

Comment:
this will fail all the different language implementations of purl-spec right now. 

the actual version output is `797174552AE47F449AB70B684CABCB6603E5E85E` for packageurl-go and packageurl-python atm

Is this expected? Should versions be downcased?

Reply:
@noqcks - that sounds reasonable. PR: https://github.com/package-url/purl-spec/pull/213